proposals/4039-widget-api-media.md

```diff
@@ -1,28 +1,62 @@
-# MSC4039: Access the Content repository with the Widget API
-
-[MSC2762](https://github.com/matrix-org/matrix-spec-proposals/pull/2762) specifies a Widget API that
-is able to receive events from and write events to the room that is loaded in the client. But beside
-the room events, homeservers also manage media content for the users. While widgets can already
-download media, given they acquired the URL of the homeserver, they can't upload new media into the
-room. This would be useful to post images or attachments, but also to upload larger files that can't
-be stored in a room event, like a whiteboard that can easily grow larger than 64kb.
-
-This proposal aims to bring the functionality of uploading media to the content repository into the
-widget specification. It should provide the same features that the
-[“Upload Content” endpoint](https://spec.matrix.org/v1.7/client-server-api/#post_matrixmediav3upload)
-of the Client-Server API provides. 
+# MSC4039: Access the Content Repository With the Widget API
+
+[MSC1236][MSC1236], [MSC2762][MSC2762], etc. specify a Widget API that is able to receive events from and write events
+to the room that is loaded in the client. Besides sending pure text events, it is possible to attach media and other
+files to events. The two concepts are consciously separated: Files are stored in a "[content repository][content]"
+(usually attached to the homeserver) and are addressable through a "[Matrix Content (`mxc://`) URI][mxc]". This URI can
+in turn be used in an event to reference a file.
+
+While prior to [MSC3916][MSC3916] and deprecation in Matrix v1.11, widgets could already download media via the
+unauthenticated download endpoint, provided they acquired the URL of the homeserver, the widget API does not currently
+contain a way to authenticate with the upload and new download endpoints nor does it otherwise expose these functions
+from the client hosting the widget to the widget itself. This would be useful to post images or attachments, but also to
+upload larger data that can't easily be stored in a room event, like a whiteboard's content that can easily grow larger
```

Comment:
I would keep this more general, something like

„but also to upload larger data that can't easily be stored in a room event exceeding the 64kb event size limit“

Reply:
this is a leftover from the original MSC; I think illustrating the point preempts questions such as "whatever could be so big!" but am also fine with changing it :shrug: